supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/EndpointParser.java

```diff
@@ -0,0 +1,217 @@
+                DeleteMapping.class.getSimpleName(), PatchMapping.class.getSimpleName(), RequestMapping.class.getSimpleName());
+        List<String> filesFailedToParse = new ArrayList<>();
+
+        for (String filePath : filePaths) {
+            CompilationUnit compilationUnit;
+            try {
+                compilationUnit = StaticJavaParser.parse(new File(filePath));
+            }
+            catch (Exception e) {
+                filesFailedToParse.add(filePath);
+                continue;
+            }
+
+            List<ClassOrInterfaceDeclaration> classes = compilationUnit.findAll(ClassOrInterfaceDeclaration.class);
+            for (ClassOrInterfaceDeclaration javaClass : classes) {
+                List<EndpointInformation> endpoints = new ArrayList<>();
+                final String classRequestMappingString = extractClassRequestMapping(javaClass, httpMethodClasses);
+
+                endpoints.addAll(extractAnnotationPathValues(javaClass, httpMethodClasses, classRequestMappingString));
+
+                if (!endpoints.isEmpty()) {
+                    endpointClasses.add(new EndpointClassInformation(javaClass.getNameAsString(), classRequestMappingString, endpoints));
+                }
+            }
+
+        }
+
+        printFilesFailedToParse(filesFailedToParse);
+
+        writeEndpointsToFile(endpointClasses);
+    }
+
+    /**
+     * Extracts endpoint information from the methods of a given class declaration.
+     *
+     * This method iterates over the methods of the provided class and their annotations.
+     * If an annotation matches one of the specified HTTP method annotations, it extracts
+     * the path values from the annotation and creates EndpointInformation objects for each path.
+     *
+     * @param javaClass                 the class declaration to extract endpoint information from
+     * @param httpMethodClasses         a set of HTTP method annotation class names
+     * @param classRequestMappingString the class-level request mapping string
+     * @return a list of EndpointInformation objects representing the extracted endpoint information
+     */
+    private static List<EndpointInformation> extractAnnotationPathValues(ClassOrInterfaceDeclaration javaClass, Set<String> httpMethodClasses, String classRequestMappingString) {
+        return javaClass.getMethods().stream().flatMap(method -> method.getAnnotations().stream().filter(annotation -> httpMethodClasses.contains(annotation.getNameAsString()))
+                .flatMap(annotation -> extractPathsFromAnnotation(annotation).stream()
+                        .map(path -> new EndpointInformation(classRequestMappingString, method.getNameAsString(), annotation.getNameAsString(), path, javaClass.getNameAsString(),
+                                method.getBegin().get().line, method.getAnnotations().stream().map(AnnotationExpr::toString).collect(Collectors.toList())))))
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                                method.getBegin().get().line, method.getAnnotations().stream().map(AnnotationExpr::toString)..toList()))))
                .toList();
```
You might have to fix the line break due to this change